tests/test_mace_fitting.py

```diff
@@ -0,0 +1,58 @@
+import yaml, os
+import time
+from pathlib import Path
+import mace, json
+from wfl.fit.mace import fit
+from wfl.configset import ConfigSet
+import ase.io
+
+# This part is used to detect the location of 'run_train.py' file. 
+mace_dist_dir = list(Path(mace.__file__).parent.parent.glob("mace-*"))[0]
+jsonfile = str(Path(mace_dist_dir / "direct_url.json"))
+with open(jsonfile) as f:
+    d = json.load(f)
+script_dir = d["url"].split(":")[-1][2:]
+mace_fit_cmd = f"python {script_dir}/scripts/run_train.py"
+
+
+def test_mace_fit_from_list(request, tmp_path, monkeypatch, mace_fit_cmd=mace_fit_cmd):
+
+    monkeypatch.chdir(tmp_path)
+    print('getting fitting data from ', request.path)
+    print("here :  ", request.path)
+	
+    parent_path = request.path.parent
+    params_file_path = parent_path / 'assets' / 'mace_fit_parameters.yaml'
+    mace_fit_params = yaml.safe_load(params_file_path.read_text())
+    filepath = parent_path / 'assets' / 'B_DFT_data.xyz'
+    fitting_configs = ConfigSet(ase.io.read(filepath, ":"))
+
+    t0 = time.time()
+    fit(fitting_configs, "test", mace_fit_params, mace_fit_cmd=mace_fit_cmd, run_dir=".")
+    t_run = time.time() - t0
+
+    assert (tmp_path / "test.model").exists()
+    assert (tmp_path / "test.model").stat().st_size > 0
+
+
+def test_mace_fit(request, tmp_path, monkeypatch, mace_fit_cmd=mace_fit_cmd):
+
+    monkeypatch.chdir(tmp_path)
+    print('getting fitting data from ', request.path)
+    print("here :  ", request.path)
+
+    parent_path = request.path.parent
+    fit_config_file = parent_path / 'assets' / 'B_DFT_data.xyz'
+    params_file_path = parent_path / 'assets' / 'mace_fit_parameters.yaml'
+    mace_fit_params = yaml.safe_load(params_file_path.read_text())
+    mace_fit_params["train_file"] = fit_config_file
```

Comment:
It's going to ignore the `"train_file"` dict entry anyway, right? We should either test without it set, or with it set to the wrong thing, to test the overriding.

Reply:
It's true that dict assignment is ignored anyway if `fitting_config` is initiated from an already writtne file. 
pytest works either way whether I set dict entry with "train_file" or not.

Reply:
Yes, but if someone breaks the ConfigSet single file writing code in the mace fit wrapper, and you pass the correct file, the pytest could still pass, even though the code that writes to a file is broken.

Reply:
I don't understand the situation of breaking the ConfigSet single file. Could you elaborate a little bit on that situation?

Reply:
The point of the tests is to make sure the code is working. If you write a test that passes even if the code is broken, that's not a useful test.  The whole point of the way the wrapper is written is that it shouldn't depend on the user passing the `"train_file"` parameter, because it's going to write the input file (if needed) by itself. If you pass that parameter, it's more likely that some time in the future something about the code that writes to a single file if needed will be broken accidentally, but this test won't fail, because it's also using a mechanism that we don't want to pass the location of the training data file.

Reply:
Okay, if I understood correctly, I should not assign "train_file" and pass it on to `mace_fit_params` dict in this pytest.
I'll remove that dict entry assignment.

Reply:
I've passed on wrong "train_file" to `mace_fit_params` but it still works well with one connected to `fitting_configs`.